shank-idl/ACCOUNTS_STRUCT_IDL.md

```
@@ -0,0 +1,130 @@
+5. **Complete IDL**: Full account information in generated IDLs for tooling consumption
+
+## Macro Integration
+
+The `ShankAccounts` macro generates the `__shank_accounts()` method that returns account metadata in this format:
+
+```rust
+Vec<(u32, &'static str, bool, bool, bool, bool, Option<String>)>
+// (index, name, writable, signer, optional_signer, optional, description)
+```
+
+This metadata should be consumed by the IDL generator to produce the expanded account list.
```

markdownlint-cli2 (0.18.1)

Files should end with a single newline character

(MD047, single-trailing-newline)